internal/controller/collection_controller.go

```
@@ -0,0 +1,374 @@
+	}
+
+	r.checkMapStatus(collection)
+
+	return nil
+}
+
+func (r *CollectionReconciler) deleteCollection(ctx context.Context, collection *v1alpha1.Collection) error {
+	// log := logf.FromContext(ctx)
+
+	// Handle collection deletion.
+	if collection.Spec.RemovalPolicy == "delete" {
```

golangci-lint (1.64.8)

SA9003: empty branch

(staticcheck)